elisp-bytecode.texi

```diff
@@ -1041,55 +1041,220 @@ bytecode interpreter.
-Call @code{<} with two arguments, comparing the two values at the top of the stack with the numerical less-than relation.
+
+@table @strong
+@item Implements:
+@math{TOS = TOS < TOS - 1}.
+@item Generated via:
+binary @code{<}.
+@item Stack effect:
+@math{-2+1}.
+@item Example:
+When lexical binding is in effect, @code{(defun lss-eg(a b) (< a b))} generates:
+@verbatim
+PC  Byte  Instruction
+ 0    1   stack-ref[1]
+ 1    1   stack-ref[1]
+ 2   87   lss
+ 3  135   return
+@end verbatim
+
+@end table
 
 @node leq
 @unnumberedsubsubsec @code{leq} (88)
 @kindex leq
-Call @code{<=} with two arguments, comparing the two values at the top of the stack with the numerical less-than-or-equals relation.
+
+@table @strong
+@item Implements:
+@math{TOS = TOS <= 1}.
+@item Generated via:
+binary @code{<=}.
+@item Stack effect:
+@math{-2+1}.
+@item Example:
+When lexical binding is in effect, @code{(defun leq-eg(a b) (<= a b))} generates:
+@verbatim
+PC  Byte  Instruction
+ 0    1   stack-ref[1]
+ 1    1   stack-ref[1]
+ 2   88   leq
+ 3  135   return
+@end verbatim
+
+@end table
 
 @node geq
 @unnumberedsubsubsec @code{geq} (89)
 @kindex geq
-Call @code{>=} with two arguments, comparing the two values at the top of the stack with the numerical less-than-or-equals relation.
+
+@table @strong
+@item Implements:
+@math{TOS = TOS >= 1}.
+@item Generated via:
+binary @code{>=}.
+@item Stack effect:
+@math{-2+1}.
+@item Example:
+When lexical binding is in effect, @code{(defun geq-eg(a b) (>= a b))} generates:
+@verbatim
+PC  Byte  Instruction
+ 0    1   stack-ref[1]
+ 1    1   stack-ref[1]
+ 2   89   geq
+ 3  135   return
+@end verbatim
+
+@end table
 
 @node diff
 @unnumberedsubsubsec @code{diff} (90)
 @kindex diff
-Call @code{-} with two arguments, subtracting the two values at the top of the stack.
+
+@table @strong
+@item Implements:
+@math{TOS = TOS - 1}.
+@item Generated via:
+binary @code{-}.
+@item Stack effect:
+@math{-2+1}.
+@item Example:
+When lexical binding is in effect, @code{(defun diff-eg(a b) (- a b))} generates:
+@verbatim
+PC  Byte  Instruction
+ 0    1   stack-ref[1]
+ 1    1   stack-ref[1]
+ 2   90   diff
+ 3  135   return
+@end verbatim
+
+@end table
 
 @node negate
 @unnumberedsubsubsec @code{negate} (91)
 @kindex negate
-Call @code{-} with one argument, negating the top-of-stack value.
+
+@table @strong
+@item Implements:
+@math{TOS = -TOS}.
+@item Generated via:
+unary @code{-}.
+@item Stack effect:
+@math{-1+1}.
+@item Example:
+When lexical binding is in effect, @code{(defun negate-eg(a) (- a))} generates:
+@verbatim
+PC  Byte  Instruction
+ 0  137   dup
+ 1   91   negate
+ 2  135   return
+@end verbatim
+
+@end table
 
 @node plus
 @unnumberedsubsubsec @code{plus} (92)
 @kindex plus
-Call @code{+} with two arguments, adding the two values at the top of the stack.
 
-@strong{Example}
-When dynamic binding is in effect, @code{(defun plus-eg(n) (+ n n ))} generates:
+@table @strong
+@item Implements:
+@math{TOS = TOS + TOS-1}.
```

Comment:
This is different syntax from what's used above.

Reply:
Also, this reads (to me) like it doesn't pop a value, when in fact it does.

Reply:
Just revised and see if this is better. 

All of this is a bit tedious and error prone. I think I got all of the things mentioned, but I might have missed stuff. Feel free just to make corrections in place. 

(And there is a lot more that we haven't gotten to). 

I think the return instruction, should be lumped in with control-flow instructions, e.g. jump and jumpif...